perf/unixbench.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Author: Santhosh G <santhog4@linux.vnet.ibm.com>
+# Copyright: 2016 IBM
+#
+# Based on code mostly written by:
+# Author : John Admanski <jadmanski@google.com>
+# copyright : 2008 Google
+
+import os
+import re
+from avocado import Test
+from avocado.utils import process
+from avocado.utils import build
+from avocado.utils import git
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import distro
+from avocado.core import data_dir
+
+
+class unixbench(Test):
+    def setUp(self):
+        sm = SoftwareManager()
+        detected_distro = distro.detect()
+        # Check for basic utilities
+        deps = ['gcc', 'git', 'make']
+        self.tmpdir = data_dir.get_tmp_dir()
+        self.build_dir = self.params.get('build_dir', default=self.tmpdir)
+        for package in deps:
+            if package == 'git' and detected_distro.name == "SuSE":
+                package = 'git-core'
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(package + ' is needed for the test to be run')
+        git.get_repo('https://github.com/kdlucas/byte-unixbench.git',
```

Comment:
I'd strongly suggest using fetch asset and `https://github.com/kdlucas/byte-unixbench/archive/master.zip`. It can be cached and downloads only the snapshot, not the whole history...